src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -63,22 +82,28 @@
 
 (defn build-routes
   [status-fns-atom]
-  (compojure/routes
-    (compojure/context "/v1" []
-      (compojure/GET "/services" []
-        (let [statuses (call-status-fns status-fns-atom)]
-          {:status 200
-           :body statuses}))
-       (compojure/GET "/services/:service-name" [service-name]
-         (if-let [service-info (get (deref status-fns-atom) service-name)]
-           (let [status (call-latest-status-fn-for-service service-info)]
-             {:status 200
-              :body (assoc status :service-name service-name)})
-           {:status 404
-            :body {:error {:type :service-not-found
-                           :message (str "No status information found for service "
-                                         service-name)}}})))))
+  (handler/api
+    (compojure/routes
+      (compojure/context "/v1" []
+        (compojure/GET "/services" [:as {params :params}]
+          (let [level (get-status-detail-level params)
+                statuses (call-status-fns (deref status-fns-atom) level)]
```

Comment:
This makes a lot of sense, and I think it's worth doing. I'm not sure that this can be done for the `update` function (which gets called by `register-status!`) since it uses `swap!` to update the atom, but it could be unless changed to create a partial that gets used with `swap!` in the service code, or it could just be moved into the service code completely. In either of those changes the tests would probably need to change, but that's not a huge deal.

In any case, I think I'll open up a separate PR for these changes.